src/login/login.component.tsx

```diff
@@ -84,23 +93,50 @@ export default function Login(props: LoginProps) {
                 value={username}
                 onChange={evt => setUsername(evt.target.value)}
                 placeholder="Username"
+                ref={usernameInputRef}
                 autoFocus
               />
             </label>
           </div>
-          <div>
+          <div
+            css={css`
+              position: relative;
+            `}
+          >
             <label>
               <input
                 css={css`
                   ${input}
                 `}
-                type="password"
+                type={showPassword ? "text" : "password"}
                 name="password"
                 value={password}
                 onChange={evt => setPassword(evt.target.value)}
                 placeholder="Password"
+                ref={passwordInputRef}
               />
             </label>
+            <button
+              className="omrs-unstyled"
+              type="button"
+              aria-label="Toggle view password text"
+              onClick={() => setShowPassword(!showPassword)}
+              css={css`
+                cursor: pointer;
+              `}
+            >
+              <svg
+                className="omrs-icon"
+                role="img"
+                css={css`
+                  top: calc(50% - 0.75rem);
```

Comment:
How did you calculate the appropriate values for top and right?

Reply:
For top, I wanted it vertically centered and the total height was 1.5rem. For right, I looked at Figma and saw that there were 12px (.75rem) padding between the edge of the input and the icon.